services/workflows-service/src/supabase/supabase.service.ts

```diff
@@ -5,38 +5,58 @@ import { ISupabaseService } from './types';
 
 @Injectable()
 export class SupabaseService implements ISupabaseService {
-  private readonly supabaseClient: SupabaseClient;
+  private readonly supabaseClient!: SupabaseClient;
   private readonly logger = new Logger(SupabaseService.name);
 
   constructor(private readonly configService: ConfigService) {
-    const supabaseUrl = this.configService.get<string>('SUPABASE_URL');
+    const telemetryEnabled = this.configService.get('TELEMETRY_ENABLED');
+    const supabaseUrl = this.configService.get<string>('TELEMETRY_SUPABASE_URL');
     const supabaseApiKey = this.configService.get<string>('TELEMETRY_SUPABASE_API_KEY');
-
-    if (!supabaseUrl || !supabaseApiKey) {
-      throw new Error('Supabase URL or API key is missing in configuration');
+    if (telemetryEnabled){
+      if (!supabaseUrl || !supabaseApiKey) {
+        throw new Error('Supabase URL or API key is missing in configuration');
+      }
+      else{
+        this.supabaseClient = createClient(
+          supabaseUrl,
+          supabaseApiKey,
+          {
+            db: { schema: 'public' },
+          }
+        );
+        this.logger.log('Supabase client initialized.');
+        const infradata = require('/tmp/infra.json');
```

Comment:
please add a comment that describe when this file or what file charge to create it